File: app/src/main/java/com/mintthursday/database/converters/ListIngredientConverter.java
package com.mintthursday.database.converters;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.mintthursday.models.Ingredient;

import java.lang.reflect.Type;
import java.util.List;

public class ListIngredientConverter {
    private final Type listType = new TypeToken<List<Ingredient>>() {
    }.getType();

    @TypeConverter
    public String toJson(List<Ingredient> list) {
        return new Gson().toJson(list, listType);
    }

    @TypeConverter
    public List<Ingredient> fromJson(String json) {
        return new Gson().fromJson(json, listType);

    }
}

File: app/src/main/java/com/mintthursday/database/converters/ListIngredientConverter.kt
package com.mintthursday.database.converters

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.mintthursday.models.Ingredient

class ListIngredientConverter {

    private val listType = object : TypeToken<List<Ingredient>>() {}.type

    @TypeConverter
    fun toJson(list: List<Ingredient>): String {
        return Gson().toJson(list, listType)
    }

    @TypeConverter
    fun fromJson(json: String): List<Ingredient> {
        return Gson().fromJson(json, listType)
    }
}
